src/scenario.c

```diff
@@ -159,7 +159,7 @@ void scenario_begin()
 	mainWindow->saved_view_x = gSavedViewX;
 	mainWindow->saved_view_y = gSavedViewY;
 
-	uint8 zoomDifference = gSavedViewZoom - mainWindow->viewport->zoom;
+	int8_t zoomDifference = gSavedViewZoom - mainWindow->viewport->zoom;
```

Comment:
While in most cases it would be, `ssize_t` is only guaranteed to be able to store `-1`